cpp/tests/experimental/spatial/linestring_distance_test.cu

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuspatial/error.hpp>
+#include <cuspatial/experimental/linestring_distance.cuh>
+#include <cuspatial/experimental/type_utils.hpp>
+#include <cuspatial/utility/vec_2d.hpp>
+
+#include <rmm/device_uvector.hpp>
+#include <rmm/device_vector.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/counting_iterator.h>
+
+namespace cuspatial {
+namespace test {
+
+using namespace cudf;
+using namespace cudf::test;
+
+template <typename T>
+struct PairwiseLinestringDistanceTest : public BaseFixture {
+};
+
+// float and double are logically the same but would require seperate tests due to precision.
+using TestTypes = ::testing::Types<float, double>;
+TYPED_TEST_CASE(PairwiseLinestringDistanceTest, TestTypes);
+
+TYPED_TEST(PairwiseLinestringDistanceTest, FromSeparateArrayInputs)
+{
+  using T       = TypeParam;
+  using CartVec = std::vector<cartesian_2d<T>>;
+
+  auto a_cart2d = rmm::device_vector<cartesian_2d<T>>{
+    CartVec({{0.0f, 0.0f}, {1.0f, 0.0f}, {2.0f, 0.0f}, {3.0f, 0.0f}, {4.0f, 0.0f}})};
+  auto b_cart2d = rmm::device_vector<cartesian_2d<T>>{
+    CartVec({{0.0f, 1.0f}, {1.0f, 1.0f}, {2.0f, 1.0f}, {3.0f, 1.0f}, {4.0f, 1.0f}})};
+  auto offset = rmm::device_vector<int32_t>{std::vector<int32_t>{std::vector<int32_t>{0}}};
```

Comment:
Why do we need this double-nested `std::vector` construction?

Reply:
Not needed. Fixed.